src/qililab/instruments/qblox/qblox_module.py

```diff
@@ -608,10 +668,16 @@ def _set_gain_imbalance(self, value: float | str | bool, sequencer_id: int):
         """
 
         self._get_sequencer_by_id(id=sequencer_id).gain_imbalance = float(value)
-        self.device.sequencers[sequencer_id].mixer_corr_gain_ratio(float(value))
+        if instrument_set:
+            self.device.sequencers[sequencer_id].mixer_corr_gain_ratio(float(value))
```

Comment:
```suggestion
        if self.device is not None:
            self.device.sequencers[sequencer_id].mixer_corr_gain_ratio(float(value))
```

Reply:
In the end after a discussion, we decided we are going to check it with:
`if hasattr(self, "device") and self.device is not None:`